### ISSUE TITLE
`dimnames.data.table` check for `"data.frame"` inheritance is too strict

### ISSUE THREAD

**Issue by gwerbin** [Rdatatable/data.table#1932]
When, for instance, piping a `data.table` through dplyr functions, elements are prepended to its `class` attribute. This causes the class check to needlessly fail:
```
identical(class(x), c("data.table", "data.frame"))
```

However the function doesn't require an _exact_ match to work, only that the S3 `names` method is available. It seems like this check could be implemented as
```
inherits(x, "data.frame")
```

without adverse effects.

Is this correct, or am I missing a reason why the check has to be so strict?

**Comment by MichaelChirico** [comment]
This was handled in #1678 if I'm not mistaken; please update your installation.

**Comment by mattdowle** [comment]
Thanks for fixing this @MichaelChirico.
@gwerbin His fix is in v1.9.8 (item 55 in NEWS) on CRAN last week.
